src/librustc_trans/trans/_match.rs

```diff
@@ -1399,12 +1418,20 @@ fn trans_match_inner<'blk, 'tcx>(scope_cx: Block<'blk, 'tcx>,
     // `compile_submatch` works one column of arm patterns a time and
     // then peels that column off. So as we progress, it may become
     // impossible to tell whether we have a genuine default arm, i.e.
-    // `_ => foo` or not. Sometimes it is important to know that in order
-    // to decide whether moving on to the next condition or falling back
-    // to the default arm.
+    //      _ => foo
+    //      (_, _) => foo
+    // or not. Sometimes it is important to know that in order to decide
+    // whether moving on to the next arm or falling back to the default
+    // one.
+    fn all_wild_pat(pat: &ast::Pat) -> bool {
+        match pat.node {
+            ast::PatWild(ast::PatWildSingle) => true,
+            ast::PatTup(ref pats) => pats.iter().all(|p| all_wild_pat(&**p)),
+            _ => false,
+        }
+    }
```

Comment:
I think this should be made more general by checking the usefulness of a wild pattern (`PatWildSingle`) against a matrix of a single pattern `pat`, much like we check reachability.